be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/RollCallValidator.scala

```diff
@@ -34,38 +38,33 @@ case object RollCallValidator extends MessageDataContentValidator {
     }
   }
 
-  // TODO check that this is correct (correct validations)
-  def validateOpenRollCall(rpcMessage: JsonRpcRequest): GraphMessage = {
-    def validationError(reason: String): PipelineError = super.validationError(reason, "OpenRollCall", rpcMessage.id)
+  def validateOpenRollCall(rpcMessage: JsonRpcRequest, validatorName: String = "OpenRollCall"): GraphMessage = {
+    def validationError(reason: String): PipelineError = super.validationError(reason, validatorName, rpcMessage.id)
 
     rpcMessage.getParamsMessage match {
       case Some(message: Message) =>
         val data: OpenRollCall = message.decodedData.get.asInstanceOf[OpenRollCall]
 
-        if (!validateTimestampStaleness(data.start)) {
-          Right(validationError(s"stale 'start' timestamp (${data.start})"))
+        val laoId: Hash = rpcMessage.extractLaoId
+        // Note: this id doesnt follow the standard sh256(string, hash, timestamp, string) other events do;
+        // this is why generateValidationId cannot be used
+        val expectedRollCallId: Hash = Hash.fromStrings(
+          EVENT_HASH_PREFIX, laoId.toString, data.opens.toString, data.opened_at.toString
+        )
+
+        if (!validateTimestampStaleness(data.opened_at)) {
+          Right(validationError(s"stale 'opened_at' timestamp (${data.opened_at})"))
+        } else if (expectedRollCallId != data.opens) {
+          Right(validationError("unexpected id 'opens'"))
         } else {
           Left(rpcMessage)
         }
       case _ => Right(validationErrorNoMessage(rpcMessage.id))
     }
   }
 
-  // TODO check that this is correct (correct validations)
   def validateReopenRollCall(rpcMessage: JsonRpcRequest): GraphMessage = {
-    def validationError(reason: String): PipelineError = super.validationError(reason, "ReopenRollCall", rpcMessage.id)
-
-    rpcMessage.getParamsMessage match {
-      case Some(message: Message) =>
-        val data: ReopenRollCall = message.decodedData.get.asInstanceOf[ReopenRollCall]
-
-        if (!validateTimestampStaleness(data.start)) {
-          Right(validationError(s"stale 'start' timestamp (${data.start})"))
-        } else {
-          Left(rpcMessage)
-        }
-      case _ => Right(validationErrorNoMessage(rpcMessage.id))
-    }
+    validateOpenRollCall(rpcMessage, "ReopenRollCall")
```

Comment:
why dropping the other checks?

Reply:
The checks are still there. They are the same as openRollCall. Thus, ReopenRollCall calls the validator of openRollCall (line 67)

Reply:
Oh, good. Same thing I ended up doing in TypeScript 😅